elide-graphql/src/main/java/com/yahoo/elide/graphql/parser/GraphQLEntityProjectionMaker.java

```diff
@@ -0,0 +1,348 @@
+        if (!(nodeSelection instanceof Field)) {
+            throw new InvalidEntityBodyException("Node selection must be a graphQL field.");
+        }
+
+        if (!"node".equals(((Field) nodeSelection).getName())) {
+            throw new InvalidEntityBodyException("GraphQL 'edges' field selection must have 'node'.");
+        }
+
+        return ((Field) nodeSelection).getSelectionSet();
+    }
+
+    /**
+     * Add a graphQL {@link Selection} to an {@link EntityProjection}
+     *
+     * @param fieldSelection field/fragment to add
+     * @param parentProjection projection that has this field/fragment
+     */
+    private void addSelection(Selection fieldSelection, final EntityProjection parentProjection) {
+        if (fieldSelection instanceof FragmentSpread) {
+            addFragment((FragmentSpread) fieldSelection, parentProjection);
+        } else if (fieldSelection instanceof Field) {
+            addField((Field) fieldSelection, parentProjection);
+        } else {
+            throw new InvalidEntityBodyException(
+                    String.format("Unsupported selection type {%s}.", fieldSelection.getClass()));
+        }
+    }
+
+    /**
+     * Resolve a graphQL {@link FragmentSpread} into {@link Selection}s and add them to an {@link EntityProjection}
+     *
+     * @param fragment graphQL fragment
+     * @param parentProjection entity projection that contains this fragment
+     */
+    private void addFragment(FragmentSpread fragment, EntityProjection parentProjection) {
+        String fragmentName = fragment.getName();
+        if (!fragmentMap.containsKey(fragmentName)) {
+            throw new InvalidEntityBodyException(String.format("Unknown fragment {%s}.", fragmentName));
+        }
+
+        FragmentDefinition fragmentDefinition = fragmentMap.get(fragmentName);
+
+        // type name in type condition of the Fragment must match the entity projection type name
+        if (parentProjection.getName().equals(fragmentDefinition.getTypeCondition().getName())) {
+            fragmentDefinition.getSelectionSet().getSelections()
+                    .forEach(selection -> addSelection(selection, parentProjection));
+        }
+    }
+
+    /**
+     * Add a new graphQL {@link Field} into an {@link EntityProjection}
+     *
+     * @param field graphQL field
+     * @param parentProjection entity projection that contains this field
+     */
+    private void addField(Field field, EntityProjection parentProjection) {
+        Class<?> parentType = parentProjection.getType();
+        String fieldName = field.getName();
+
+        // this field would either be a relationship field or an attribute field
+        if (entityDictionary.getRelationshipType(parentType, fieldName) != RelationshipType.NONE) {
+            // handle the case of a relationship field
+            final Class<?> relationshipType =
+                    entityDictionary.getParameterizedType(parentType, fieldName) == null
```

Comment:
This check for null shouldn't be necessary.  getParameterized type will return the correct type regardless of whether the type is actually parameterized.